Extension/src/id.ts

```diff
@@ -0,0 +1,64 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import { execChildProcess } from './common';
+import { isWindows } from './constants';
+import { logLanguageServerEvent } from './telemetry';
+
+/**
+ * Hash the MAC addresses on the machine (Windows-only) and log telemetry.
+ */
+export async function logMachineIdMappings(): Promise<void> {
+    if (!isWindows) {
+        return;
+    }
+
+    const macAddresses = await getMacAddresses();
+
+    // The first MAC address is the one Visual Studio uses
+    const primary = await getMachineId(macAddresses.shift());
+    if (primary) {
+        logLanguageServerEvent('machineIdMap', {primary});
+    }
+
+    // VS Code uses os.networkInterfaces() which has different sorting and availability,
+    // but all MAC addresses are returned by getmac.exe. The ID VS Code uses may change
+    // based on changes to the network configuration. Log the extras so we can assess
+    // how frequently this impacts the machine id.
+    for (const macAddress of macAddresses) {
+        const additional = await getMachineId(macAddress);
+        if (additional) {
+            logLanguageServerEvent('machineIdMap', {additional});
```

Comment:
It seems like the additional and primary mac addresses are being logged the same -- is that okay or would there be a need to distinguish between the two?

Reply:
i.e. it's a simplification of { primary: primary }, right?

Reply:
Yeah, that's what it's doing (I found an existing example `telemetry.logLanguageServerEvent('configurationSelection', { action, `.